src/languages/swift.js

```diff
@@ -224,24 +242,46 @@ export default function(hljs) {
     USER_DEFINED_ATTRIBUTE
   ];
 
+  // https://docs.swift.org/swift-book/ReferenceManual/Types.html
   const TYPE = {
-    className: 'type',
-    begin: '\\b[A-Z][\\w\u00C0-\u02B8\']*',
-    relevance: 0
-  };
-  // slightly more special to swift
-  const OPTIONAL_USING_TYPE = {
-    className: 'type',
-    begin: '\\b[A-Z][\\w\u00C0-\u02B8\']*[!?]',
-    relevance: 0
+    begin: lookahead(/\b[A-Z]/),
+    relevance: 0,
+    contains: [
+      { // Common Apple frameworks, for relevance boost
+        className: 'type',
+        begin: concat(/(AV|CA|CF|CG|CI|CL|CM|CN|CT|MK|MP|MTK|MTL|NS|SCN|SK|UI|WK|XC)/, Swift.identifierCharacter, '+')
+      },
+      { // Type identifier
+        className: 'type',
+        begin: Swift.typeIdentifier,
+        relevance: 0
+      },
+      { // Optional type
+        begin: /[?!]+/,
+        relevance: 0
+      },
+      { // Variadic parameter
+        begin: /\.\.\./,
+        relevance: 0
+      },
+      { // Protocol composition
+        begin: concat(/\s+&\s+/, lookahead(Swift.typeIdentifier)),
+        relevance: 0
+      }
+    ]
   };
-  const BLOCK_COMMENT = hljs.COMMENT(
-    '/\\*',
-    '\\*/',
-    {
-      contains: [ 'self' ]
-    }
-  );
+  // Generic arguments
+  TYPE.contains.push({
+    begin: /</,
+    end: />/,
+    keywords: KEYWORDS,
+    contains: [
+      ...KEYWORD_MODES,
+      ...ATTRIBUTES,
+      OPERATOR_GUARD,
+      TYPE
```

Comment:
'self'?

Reply:
Nevermind, it's nested.

Reply:
Yeah, the nesting can get pretty complex.